packages/chain-bridge/src/adapters/LayerZeroAdapter.sol

```diff
@@ -273,6 +490,47 @@ contract LayerZeroAdapter is
                         INTERNAL FUNCTIONS
     //////////////////////////////////////////////////////////////*/
 
+    /**
+     * @notice Validates asset and amount, returns OFT contract
+     * @param asset The ERC20 token address
+     * @param amount The amount to transfer
+     * @return oft The OFT contract address for this asset
+     */
+    function _getOFTForAsset(
+        address asset,
+        uint256 amount
+    ) internal view returns (address oft) {
+        oft = oftForToken[asset];
+        if (oft == address(0)) revert UnsupportedAsset();
+        if (amount == 0) revert InvalidParams();
+    }
+
+    /**
+     * @notice Creates OFT SendParam struct
+     * @param dstEid Destination endpoint ID (pre-computed)
+     * @param dstAdapter Destination adapter address
+     * @param amount Amount to send in local decimals
+     * @param composeMsg Optional compose message
+     * @return sendParam Configured SendParam struct
+     */
+    function _createOFTSendParam(
+        uint32 dstEid,
+        address dstAdapter,
+        uint256 amount,
+        bytes memory composeMsg
+    ) internal pure returns (SendParam memory) {
+        return
+            SendParam({
+                dstEid: dstEid,
+                to: dstAdapter.toBytes32(),
+                amountLD: amount,
+                minAmountLD: amount, // Exact amount, no slippage
+                extraOptions: bytes(""),
```

Comment:
I believe it's missing handling for the compose additinal gas as per : https://docs.layerzero.network/v2/developers/evm/oft/quickstart#execution-options-for-composed-messages